src/components/UserDetails/UserDetails.jsx

```diff
@@ -0,0 +1,241 @@
+      setErrorMessage(firebaseErrorMsg);
+    }
+  };
+
+  const handleCancel = () => {
+    reset({
+      firstName: user.firstName,
+      lastName: user.lastName,
+      phoneNumber: user.phoneNumber,
+      newPassword: '',
+      confirmPassword: '',
+    });
+    setCanEditForm(false);
+  };
+
+  return (
+    <Flex minH="50vh" align="center" justify="center">
+      <Stack className={styles['user-details-stack']}>
+        <Heading align="center">User Profile</Heading>
+        {errorMessage && <Box>{errorMessage}</Box>}
+        <form onSubmit={handleSubmit(updateUserDetails)}>
+          <FormControl isRequired>
+            <FormLabel className={styles['user-details-form-label']}>First Name</FormLabel>
+            <Input
+              id="first-name"
+              placeholder="Enter first name"
+              {...register('firstName')}
+              isRequired
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.firstName?.message}</Box>
+          </FormControl>
+          <FormControl isRequired>
+            <FormLabel className={styles['user-details-form-label']}>Last Name</FormLabel>
+            <Input
+              id="last-name"
+              placeholder="Enter last name"
+              {...register('lastName')}
+              isRequired
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.lastName?.message}</Box>
+          </FormControl>
+          <FormControl isRequired>
+            <FormLabel className={styles['user-details-form-label']}>Phone Number</FormLabel>
+            <Input
+              type="tel"
+              id="phone-number"
+              placeholder="Enter phone number"
+              {...register('phoneNumber')}
+              isRequired
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.phoneNumber?.message}</Box>
+          </FormControl>
+          <FormLabel className={styles['user-details-form-label']}>Email</FormLabel>
+          <Input
+            type="email"
+            id="email"
+            placeholder="Enter email"
+            value={user.email}
+            {...register('email')}
+            isRequired
+            isDisabled
+          />
+          <Box className={styles['error-box']}>{errors.email?.message}</Box>
+          <FormControl isRequired={getValues('newPassword')?.length > 0}>
+            <FormLabel className={styles['user-details-form-label']}>New Password</FormLabel>
+            <Input
+              type="password"
+              id="new-password"
+              placeholder="Enter new password"
+              {...register('newPassword')}
+              isRequired={getValues('newPassword')?.length > 0}
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.newPassword?.message}</Box>
```

Comment:
Can we add an additional field autoComplete="new-password" so that it does not populate the field automatically?